src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.tsx

```diff
@@ -31,11 +32,15 @@ export const RespondentDataSummary = () => {
     if (selectedActivity) return;
 
     const selectedActivityByDefault =
-      getActivityWithLatestAnswer(summaryActivities) || summaryActivities[0];
+      getActivityWithLatestAnswer(summaryActivities) || summaryActivities?.[0];
 
     if (!selectedActivityByDefault) return;
 
+    const { startDate, endDate } = getOneWeekDateRange(selectedActivityByDefault.lastAnswerDate);
     setValue('selectedActivity', selectedActivityByDefault);
+    startDate && setValue('startDate', startDate);
+    endDate && setValue('endDate', endDate);
```

Comment:
Could you please move this logic in rows 39-42 and in ReportMenu.tsx (rows 26-29) into a separate function?